### PR TITLE
Updated mariadb to 10.6.4-r2 and alpine to 3.15

### PR DESCRIPTION
**Pull request by svasek**
Hi @jbergstroem 
First, I'd like to thank you for your work. I use your docker image for my web.

I've updated it to the most recent version of Mariadb and Alpine. 

FYI "/usr/bin/mariadb*" files are binaries and "/usr/bin/mysql*" files are symlinks so I've included them as well.

### CODE REVIEW

**Comment by jbergstroem** [comment]
@svasek thanks a lot for your contribution. My idea was to move to a multi-versioned repo but work/other open source work has kind of taken my focus. Would this be something beneficial to you? Would you be interested in helping out even?

A bit more info here: https://github.com/jbergstroem/mariadb-alpine/issues/35

I can elborate on my thoughts in detail if it interests you.

**Comment by svasek** [comment]
VERSION=bb93136 sh/run-tests.bash
 ✓ should output mysqld version
 ✓ start a default server with InnoDB and no password
 ✓ start a server without a dedicated volume (issue #1)
 ✓ start a server without InnoDB
 ✓ start a server with a custom root password
 ✓ start a server with a custom database
 ✓ start a server with a custom database, user and password
 ✓ should allow to customize the database charset
 ✓ should allow to customize the database collation
 ✓ verify that binary logging is turned off
 ✓ should allow a user to pass a custom config
 ✓ should import a .sql file and execute it
 ✓ should import a compressed file and execute it
 ✓ should execute an imported shell script

14 tests, 0 failures

File: my.cnf [hunk 1]
```diff
@@ -2,5 +2,6 @@
 innodb_buffer_pool_size = 10M
 innodb_log_buffer_size = 512K # default / 2
 innodb_log_file_size = 8M
+aria_log_file_size = 8M
```
Comment (jbergstroem) [inline my.cnf:5]: good one, thanks for this

**Comment by jbergstroem** [comment]
> From my perspective, the multi-versioned repo doesn't make sense for this project as there is not something like LTS Alpine. The multi-versioned repo would make sense for full-featured MariaDB based on Ubuntu or UBI, IMHO.

The goal I wanted to achieve was to allow people to stick with a mariadb/mysql version they are used to. As you know, some things changes in mysql land as majors hit. I can only agree with the fact that alpine isn't the best place to build your own LTS ground but 3.12 indeed got some backports, even out of "LTS" alpine versioning. 

> The purpose of this image is to be minimalistic so it's fit for development or lightweight web. Those attributes are what make this project unique. So, we/you should probably focus only on the latest version for security reasons. Would be nice to have old versions in branches of course.

Yes, the goal is to achieve a very out-of-the-box working but strong emphasis on size container.

> Please, correct me if I'm wrong ;-)

Nothing wrong! Seeing how this is still relatively small; every feedback/alignment matters. Will most likely merge as-is tomorrow and revisit the tagging.

(also, good to hear from more users; bring interest back into fixing a lot of the things I've raised as points to improve)

**Comment by svasek** [comment]
> One remaining question: why add the symlinks?

1. It was not necessary to change any script
2. They have small size, so why not

**Comment by svasek** [comment]
I've added a new commit where I've changed the `default_storage_engine` from `MyISAM` to `Aria` when `SKIP_INNODB=1`.

**Comment by jbergstroem** [comment]
> > One remaining question: why add the symlinks?
> 
> 1. It's safe as it wasn't necessary to make changes in a scripts
> 2. They have small size, so why not

My 2c is that they bring no benefit to the user (why would they invoke them?) at the cost of extra size. Could you update the scripts and use a [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) message style? With that I'll merge! Thanks a lot.

**Comment by svasek** [comment]
> My 2c is that they bring no benefit to the user (why would they invoke them?) at the cost of extra size

Alpine linux is full of symlinks as it's based on busybox so all system command are served via symlinks. 
So, 3 more symlinks are not an issue. Besides that one symlink has size only `12B` ;-) 

This pull request is not about refactoring. We can let it for next one IMHO.

**Comment by jbergstroem** [comment]
> Sorry I don't understand. Should I delete both commits and create new with conventional commit messages?

Sorry, took the weekend off. I'm happy to land all commits in this PR but lets just change the commit titles/bodies a bit. The easiest way is usually just rebasing locally `git rebase -i master` and then just using `r` for reword. The editor should allow you to only edit the commit message.

> This pull request is not about refactoring. We can let it for next one IMHO.

Right; ok! Up to you. Either here or in a new PR. Thanks a lot for helping out.

**Comment by svasek** [comment]
So, I've 

> Could you update the scripts and use a [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) message style? With that I'll merge!

I've updated commit messages.

**Comment by jbergstroem** [comment]
> > Could you update the scripts and use a [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) message style? With that I'll merge!
> 
> I've just updated commit messages. Please take a look.

Looks good. Will land if tests pass.

**Review by jbergstroem — APPROVED** [review]

**Pull request merged**